src/frontend/src/pages/AdminToolsPage/BOMConfig/ManufacturerTable.tsx

```diff
@@ -1,35 +1,115 @@
-import { TableRow, TableCell, Box } from '@mui/material';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { Box, Grid, IconButton, TableCell, TableRow } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Manufacturer } from 'shared';
 import LoadingIndicator from '../../../components/LoadingIndicator';
+import { NERButton } from '../../../components/NERButton';
+import { useDeleteManufacturer, useGetAllManufacturers } from '../../../hooks/bom.hooks';
+import { useToast } from '../../../hooks/toasts.hooks';
 import { datePipe } from '../../../utils/pipes';
 import ErrorPage from '../../ErrorPage';
-import { NERButton } from '../../../components/NERButton';
-import { useState } from 'react';
 import AdminToolTable from '../AdminToolTable';
-import { useGetAllManufacturers } from '../../../hooks/bom.hooks';
 import CreateManufacturerModal from './CreateManufacturerFormModal';
+import ManufacturerDeleteModal from './ManufacturerDeleteModal';
+
+interface ManufacturerDeleteButtonProps {
+  name: string;
+  onDelete: (name: string) => void;
+}
 
 const ManufacturerTable: React.FC = () => {
   const {
-    data: manufactureres,
-    isLoading: manufactureresIsLoading,
+    data: manufacturers,
+    isLoading: manufacturersIsLoading,
     isError: manufacturersIsError,
     error: manufacturersError
   } = useGetAllManufacturers();
   const [createModalShow, setCreateModalShow] = useState<boolean>(false);
+  const [manufacturerList, setManufacturerList] = useState<Manufacturer[]>([]);
+  const { mutateAsync } = useDeleteManufacturer();
+  const toast = useToast();
 
-  if (!manufactureres || manufactureresIsLoading) {
+  useEffect(() => {
+    if (manufacturers) {
+      setManufacturerList(manufacturers);
+    }
+  }, [manufacturers]);
+
+  if (!manufacturers || manufacturersIsLoading) {
```

Comment:
you shouldn't need this useEffect, the query invalidation in the hooks should handle updating the list of manufacturers